be/src/olap/tablet.cpp

```diff
@@ -826,6 +832,33 @@ void Tablet::_print_missed_versions(const std::vector<Version>& missed_versions)
     LOG(WARNING) << ss.str();
 }
 
+bool Tablet::_drop_rowset_if_version_conflict(const RowsetSharedPtr& rowset) {
+    if (rowset == nullptr) {
+        return false;
+    }
+    Version insert_version = {rowset->start_version(), rowset->end_version()};
+    Version max_version = _tablet_meta->max_version();
+    if (insert_version != max_version) {
+        return false;
+    }
+
+    RowsetSharedPtr exist_rs = get_rowset_by_version(insert_version);
+    // if a rs is created by doris 0.10 exists, remove it.
+    if (exist_rs != nullptr) {
+        vector<RowsetSharedPtr> to_add;
+        vector<RowsetSharedPtr> to_delete;
+        to_delete.push_back(exist_rs);
+        modify_rowsets(to_add, to_delete);
+        std::stringstream ss;
+        ss << "Remove rs " << exist_rs->rowset_id().to_string();
```

Comment:
LOG(WARNING) << "Remove rs " << exist_rs->rowset_id().to_string()
                             << " in " << full_name()
                             << " for version conflict with " << rowset->rowset_id().to_string()
There is no necessity to use std::stringstream.